pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,402 @@
+		// NOTE: This additional status check was handled to avoid duplicate cache processing for the same PodName and PodNamespace.
+		// PodSecondaryIface shows the status of secondary network interface configuration. If already configured, avoid processing it again (As per the current implementation,
+		// we do not have full support for Pod update event handle (i'e, Pod annotation changes etc.,).
+		// TBD: Impelment Linux interface configuration API to look for secondary network interface in container scope and replace PodSecondaryIface check with the new API (Ideal
+		// way to handle).
+		if containerInfo.PodCNIStatus != podDelComplete && containerInfo.PodSecondaryIface != false {
+			// Update Virtual Network configuration (retrieved from network-attach-definition) at CNI cache. This is required to release IP address at whereabouts DB.
+			containerInfo.NetworkInfo = cniconfig
+			containerInfo.ContainerIfDev = netinfo.InterfaceName
+			pc.podCache.UpdateCNIConfigInfo(containerInfo)
+			if netinfo.InterfaceType == sriovInterfaceType {
+				cmdArgs = &invoke.Args{Command: string("ADD"), ContainerID: containerInfo.ContainerID,
+					NetNS: containerInfo.ContainerNetNS, IfName: netinfo.InterfaceName,
+					Path: cniPath}
+
+				ipamResult, err = ipam.GetIPAMSubnetAddress(cniconfig, cmdArgs)
+				if err != nil {
+					return errors.New("secondary network IPAM failed")
+				}
+
+				result := &current.Result{CNIVersion: containerInfo.CNIVersion}
+				result.IPs = ipamResult.IPs
+				result.Routes = ipamResult.Routes
+				// Set result.Interface to container interface.
+				for _, ip := range result.IPs {
+					ip.Interface = current.Int(1)
+				}
+				// Configure SRIOV as a secondary network interface
+				if err = pc.configureSriovAsSecondaryInterface(pod, netinfo, containerInfo, result); err != nil {
+					// SRIOV interface creation failed. Free allocated IP address
+					if ipamerr = ipam.DelIPAMSubnetAddress(cniconfig, cmdArgs); ipamerr != nil {
+						klog.ErrorS(err, "IPAM de-allocation failed: ", ipamerr)
+					}
+					return err
+				}
+				containerInfo.PodSecondaryIface = true
+			} else {
+				klog.ErrorS(err, "InterfaceType not supported for Pod ", klog.KObj(pod))
+			}
+		}
+	}
+	return nil
+}
+
+func (pc *PodController) configureSecondaryNetwork(pod *corev1.Pod, networklist []*SecondaryNetworkObject, podCNIInfo []*cnipodcache.CNIConfigInfo) error {
+
+	for _, netinfo := range networklist {
+		klog.InfoS("Secondary Network Information:", netinfo)
+		if len(netinfo.NetworkName) > 0 {
+			netDefCRD, err := pc.netAttachDefClient.NetworkAttachmentDefinitions(pod.Namespace).Get(context.TODO(), netinfo.NetworkName, metav1.GetOptions{})
+			if err != nil {
+				klog.ErrorS(err, "NetworkAttachmentDefinitions.Get failed", "pod", klog.KObj(pod))
+				// NetworkAttachmentDefinition not found at this time. Return error to re-queue and re-try.
```

Comment:
Here if we have a partial failure (e.g. some interfaces created, some not), are we able to handle it in retry? What will happen for interfaces that had been successfully created earlier in the retry?

Reply:
As we talked, probably we should return nil, as long as one interface is created. Probably add a comment to explain it is because we do not check and handle Pod update at this momentum.